### PR TITLE
CSV Generator and Extractor

### PR DESCRIPTION
**Pull request by sourcerer-mike**
CSV files are a good format for customers and non-developers.
To exchange translations with them a csv generator and extractor have been introduced.

### CODE REVIEW

**Comment by sourcerer-mike** [comment]
PHP 5.3 go and get your pension! :/

**Comment by mlocati** [comment]
It's not always that easy to move projects to newer PHP versions, in particular the big and widely used ones

**Comment by sourcerer-mike** [comment]
I know and that's okay. Old projects never realy switch to newer packages. Feel free to drop some support sometimes with a new major release or so ;)
Anyway I fixed it for PHP 5.3

Hope you merge it :) I really like to use it in some projects and have a better base for customers. They know Excel so they can help translating now ;)

File: src/Generators/CsvDictionary.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        //for a simple json translation dictionary, one domain is supported
+        $values = current($array);
+
+        // remove meta / header data
+        if (array_key_exists('', $values)) {
+            unset($values['']);
+        }
+
+        $tmpFile = tempnam(sys_get_temp_dir(), 'gettext_');
+        $handle = fopen($tmpFile, 'w');
```
Comment (oscarotero) [inline src/Generators/CsvDictionary.php]: Why not use ` fopen('php://temp', 'w') `? Seems more optimized than creating a real file in disk.

File: README.md [hunk 2]
```diff
@@ -196,6 +196,7 @@ $string = file_put_contents('locales/en.po', $content);
 
 The available generators are:
 
+* `Gettext\Generators\Csv` - Exports to CSV format
```
Comment (oscarotero) [inline README.md]: The generator and extractor is called CsvDictionary instead Csv :)

File: src/Generators/CsvDictionary.php (hunk 1 above)
Comment (sourcerer-mike) [inline src/Generators/CsvDictionary.php]: I don't know if others use that stream already so a file is a fail-safe method.
If `php://temp` will create a new file or stream with every fopen then this is fail-safe as well.

Anyway I change it to the "php://temp" now.

**Comment by sourcerer-mike** [comment]
I read the comments and changed the code. Hope to see an merge soon :)
Thanks for the notes!

**Comment by oscarotero** [comment]
Thanks for your contribution :+1:

**Pull request merged**